pkg/scheduler/framework/plugins/dynamicresources/allocateddevices.go

```diff
@@ -50,11 +53,21 @@ func foreachAllocatedDevice(claim *resourceapi.ResourceClaim, cb func(deviceID s
 			// Is not considered as allocated.
 			continue
 		}
-		deviceID := structured.MakeDeviceID(result.Driver, result.Pool, result.Device)
 
+		deviceID := structured.MakeDeviceID(result.Driver, result.Pool, result.Device)
+		shared := result.ShareID != nil
 		// None of the users of this helper need to abort iterating,
 		// therefore it's not supported as it only would add overhead.
-		cb(deviceID)
+		if !shared {
+			dedicatedDeviceCallback(deviceID)
+		} else {
+			sharedDeviceID := structured.MakeSharedDeviceID(deviceID, *result.ShareID)
+			sharedDeviceCallback(sharedDeviceID)
+			if result.ConsumedCapacities != nil {
+				deviceConsumedCapacity := structured.NewDeviceConsumedCapacity(deviceID, result.ConsumedCapacities)
+				consumedCapacityCallback(deviceConsumedCapacity)
+			}
+		}
```

Comment:
Flip the condition in if:
```suggestion
		if shared {
			sharedDeviceID := structured.MakeSharedDeviceID(deviceID, *result.ShareID)
			sharedDeviceCallback(sharedDeviceID)
			if result.ConsumedCapacities != nil {
				deviceConsumedCapacity := structured.NewDeviceConsumedCapacity(deviceID, result.ConsumedCapacities)
				consumedCapacityCallback(deviceConsumedCapacity)
			}
		} else {
			dedicatedDeviceCallback(deviceID)
		}
```
Or just put the `continue` after `dedicatedDeviceCallback(deviceID)`